notifications/http_test.go

```diff
@@ -122,7 +122,7 @@ func TestHTTPSink(t *testing.T) {
 		},
 		{
 			// Case where connection never goes through.
-			url:     "http://shoudlntresolve/",
+			url:     "shoudlntresolve",
```

Comment:
What is dropping the prefix here actually doing though? If this test is trying to cover a case where the connection cannot be established then I am not sure I understand this change. Could we just use a local server and either return an expected failure or deny the connection.